src/compiler/parser.ts

```diff
@@ -2635,7 +2635,7 @@ namespace Parser {
     function createIdentifier(isIdentifier: boolean, diagnosticMessage?: DiagnosticMessage, privateIdentifierDiagnosticMessage?: DiagnosticMessage): Identifier {
         if (isIdentifier) {
             identifierCount++;
-            const pos = getNodePos();
+            const pos = scanner.hasLeadingAsterisks() ? scanner.getTokenStart() : getNodePos();

```

Comment:
I feel like this is odd; why is the asterisk a part of the full start of node in the first place?

Reply:
@jakebailey Thanks for the feedback.

For similar JsDoc cases, the parser uses the following https://github.com/microsoft/TypeScript/blob/15f67e0b482faf9f6a3ab9965f3c11196bf3e99b/src/compiler/parser.ts#L9848-L9861 which references the start token position instead of the full start.

We could try changing `fullStartPos` to advance to token start directly in the `scanner`, however, this might cause breaking changes in certain scenarios - need to check...

The `import` tag parsing reuses as much logic as possible from import declarations to avoid duplication and accommodate differences. Essentially, it only requires a change in context for the scanner to skip leading asterisks and avoid deep args passing https://github.com/microsoft/TypeScript/blob/15f67e0b482faf9f6a3ab9965f3c11196bf3e99b/src/compiler/parser.ts#L8461-L8463

Since the `import specifier` uses `parseIdentifierName`, which calls `createIdentifier` and sets pos as the `fullStartPos` of the identifier, it causes an issue with the identifier range. We could switch to using `parseJSDocIdentifierName` instead of https://github.com/microsoft/TypeScript/blob/15f67e0b482faf9f6a3ab9965f3c11196bf3e99b/src/compiler/parser.ts#L8558, or modify `createIdentifier` directly. What do you think would work best for this case?

Reply:
I would think advancing it when there's an ignored asterisk is correct (what nathan said i think)